### PR TITLE
Feature/mailing list

### PR DESCRIPTION
**Pull request by maksat990rd**

### CODE REVIEW

File: src/main/java/kg/peaksoft/giftlistb6/api/MailingListApi.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@Tag(name = "Mailing list Api", description = "Admin can add mailing list")
+@RequestMapping("api/mailing-list")
+public class MailingListApi {
+
+    private final MailingListService mailingListService;
+
+    @Operation(summary = "Save mailing list", description = "Can save mailing list  ")
+    @PostMapping
+    public AllMailingListResponse saveUserInfo(@RequestBody MailingListRequest request) {
+        return mailingListService.saveMailingList(request);
```
Comment (Aiza-anarbekova) [inline src/main/java/kg/peaksoft/giftlistb6/api/MailingListApi.java:28]: почему название метода save user info?

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/AllMailingListResponse.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class AllMailingListResponse {
+
+    private Long id;
+    private String image;
+    private String name;
+    private LocalDateTime localDateTime;
+}
```
Comment (Aiza-anarbekova) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/AllMailingListResponse.java:20]: createdAt

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/MailingListResponse.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+public class MailingListResponse {
+
+    private Long id;
+    private String name;
+    private String image;
+    private String text;
+    private LocalDateTime localDateTime;
+}
```
Comment (Aiza-anarbekova) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/MailingListResponse.java:21]: createdAt

File: src/main/java/kg/peaksoft/giftlistb6/api/MailingListApi.java (hunk 1 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/api/MailingListApi.java:28]: Исправлено !

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/MailingListResponse.java (hunk 3 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/dto/responses/MailingListResponse.java:21]: Готовою

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/AllMailingListResponse.java (hunk 2 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/dto/responses/AllMailingListResponse.java:20]: Готово!

File: src/main/java/kg/peaksoft/giftlistb6/db/models/MailingList.java [hunk 4]
```diff
@@ -20,6 +22,12 @@ public class MailingList {
 
     private String name;
 
+    private String photo;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/models/MailingList.java]: Rename to image

File: src/main/java/kg/peaksoft/giftlistb6/db/models/MailingList.java [hunk 5]
```diff
@@ -20,6 +22,12 @@ public class MailingList {
...
+
     @Column(length = 10000)
     private String text;
+
+
+    private LocalDateTime createDate;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/models/MailingList.java]: Rename to createdAt

File: src/main/java/kg/peaksoft/giftlistb6/db/repositories/MailingListRepository.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.Optional;
+
+public interface MailingListRepository extends JpaRepository<MailingList, Long> {
+
+    @Query("select new kg.peaksoft.giftlistb6.dto.responses.MailingListResponse(" +
+            "m.id," +
+            "m.photo," +
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/repositories/MailingListRepository.java]: image

File: src/main/java/kg/peaksoft/giftlistb6/db/repositories/MailingListRepository.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+            "m.name," +
+            "m.text," +
+            "m.createDate)from MailingList m where m.id = :id")
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/repositories/MailingListRepository.java]: createdAt

File: src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+       List<AllMailingListResponse> list = new ArrayList<>();
+       for (MailingList mailingList : mailingLists) {
+           list.add(convertToResponse(mailingList));
+       }
+       return list;
+    }
+
+    public MailingListResponse getId(Long id) {
+         return mailingListRepository.findMailingById(id).orElseThrow(
+                ()-> new NotFoundException(String.format("Рассылка с id = %s не найдена!", id))
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java]: Рассылка с id: %s не найдена!

File: src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        MailingList mailingList = convertToEntity(request);
+        mailingListRepository.save(mailingList);
+        return convertToResponse(mailingList);
+    }
+
+    @Transactional
+    public MailingList convertToEntity(MailingListRequest request) {
+        MailingList mailingList = new MailingList();
+        mailingList.setName(request.getName());
+        mailingList.setPhoto(request.getImage());
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java]: Why setPhoto and getImage?

File: src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        mailingList.setText(request.getText());
+        mailingList.setCreateDate(LocalDateTime.now());
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java]: CreatedAt

File: src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        return mailingList;
+    }
+
+    @Transactional
+    public AllMailingListResponse convertToResponse(MailingList mailingList) {
+        AllMailingListResponse response = new AllMailingListResponse();
+        response.setId(mailingList.getId());
+        response.setName(mailingList.getName());
+        response.setImage(mailingList.getPhoto());
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java]: Why setImage and getPhoto?

File: src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+        response.setCreatAt(LocalDateTime.now());
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java]: createdAt

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/AllMailingListResponse.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class AllMailingListResponse {
+
+    private Long id;
+    private String image;
+    private String name;
+    private LocalDateTime creatAt;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/AllMailingListResponse.java]: createdAt

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/MailingListResponse.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class MailingListResponse {
+
+    private Long id;
+    private String name;
+    private String image;
+    private String text;
+    private LocalDateTime creatAt;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/MailingListResponse.java]: createdAt

**Review by i-Coderr — COMMENTED** [review]
Commented at 1 pm

File: src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java (hunk 9 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java]: entity was created with photo field, ok i edited it

File: src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java (hunk 11 above)
Comment (maksat990rd) [reply src/main/java/kg/peaksoft/giftlistb6/db/services/MailingListService.java]: because the entity was created with photo fields

**Comment by i-Coderr** [comment]
Good job!

**Pull request merged**